test/ipcFixtures.hpp

```diff
@@ -57,6 +64,21 @@ struct umfIpcTest : umf_test::test,
         test::SetUp();
         auto [pool_ops, pool_params, provider_ops, provider_params, accessor] =
             this->GetParam();
+
+        if (provider_ops == umfDevDaxMemoryProviderOps()) {
```

Comment:
Why do we need these checks here? In the `provider_devdax_memory.cpp` you can decide whether you would like to instantiate the test or not.

Reply:
Because if any of the following **environment variables** is not set: `UMF_TESTS_DEVDAX_PATH` or `UMF_TESTS_DEVDAX_SIZE` then the tests should be skipped.

Reply:
I understand your intention to check if the `UMF_TESTS_DEVDAX_PATH` and `UMF_TESTS_DEVDAX_SIZE` environments are set. But I do not think we should check for a particular memory provider here.

My suggestion is the following: instead of always instantiating this test in the `provider_devdax_memory.cpp` on the line 364 and do check here we should not instantiate the `umfIpctests` for devdax provider if the required environment variable were not set. Or you can keep the `ipcProxyPoolTestParamsList` list empty if required environment is not set.

Reply:
But it would cause multiplication of the code with checking those variables - the same code will have to be placed in each and every provider that uses `ipcFixtures.hpp` - is it really better?

Reply:
> the same code will have to be placed in each and every provider that uses ipcFixtures.hpp

Hmm, I did not get your point. This check is specific for DEVDAX provider.

Reply:
Yes, I meant for each and every pool (not provider), but if I use https://github.com/oneapi-src/unified-memory-framework/pull/845#discussion_r1827643900 it will be used only once in one file, so it will be OK.

Reply:
Done, is it OK now?